### PR TITLE
chore: Add app layout skeleton internal component

### PR DESCRIPTION
**Pull request by just-boris**
### Description

Part of a bigger project to refactor app layout and make it lazy loadable

Related links, issue #, if available: n/a

### How has this been tested?

Added a new dev page. [Click here to preview](https://d21d5uik3ws71m.cloudfront.net/components/0e4d1e6c26686e1e6277556dd28413a9438056a1/dev-pages/index.html#/light/app-layout/skeleton)

<details>
   <summary>Review checklist</summary>

_The following items are to be evaluated by the author(s) and the reviewer(s)._

#### Correctness

- _Changes include appropriate documentation updates._
- _Changes are backward-compatible if not indicated, see [`CONTRIBUTING.md`](https://github.com/cloudscape-design/components/blob/main/CONTRIBUTING.md#public-apis)._
- _Changes do not include unsupported browser features, see [`CONTRIBUTING.md`](https://github.com/cloudscape-design/components/blob/main/CONTRIBUTING.md#browsers-support)._
- _Changes were manually tested for accessibility, see [accessibility guidelines](https://cloudscape.design/foundation/core-principles/accessibility/)._

#### Security

- _If the code handles URLs: all URLs are validated through [the `checkSafeUrl` function](https://github.com/cloudscape-design/components/blob/main/src/internal/utils/check-safe-url.ts)._

#### Testing

- _Changes are covered with new/existing unit tests?_
- _Changes are covered with new/existing integration tests?_
</details>

---

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

File: src/app-layout/skeleton/index.tsx [hunk 1]
```diff
@@ -0,0 +1,46 @@
...
+  toolsWidth,
+
+  placement,
+  contentType,
+  maxContentWidth,
+  minContentWidth,
+  disableContentPaddings,
+}: AppLayoutPropsWithDefaults) {
+  // render nothing in the skeleton state
+  const placeholder = <></>;
```
Comment (just-boris) [inline src/app-layout/skeleton/index.tsx:28]: There are some plans to render some kind of grey boxes, but let's keep an empty fragment for now

File: src/app-layout/skeleton/layout.tsx [hunk 2]
```diff
@@ -0,0 +1,78 @@
...
+      <section className={styles['top-bar']}>{topBar}</section>
+      {navigation && (
+        <div className={styles['left-panel']} style={{ inlineSize: navigationWidth }}>
+          {navigation}
+        </div>
+      )}
+      <main className={styles['main-landmark']}>
+        {notifications && <div className={styles.notifications}>{notifications}</div>}
+        <div className={clsx(styles.main, { [styles['main-disable-paddings']]: disableContentPaddings })}>
+          {contentHeader && <div className={styles.contentHeader}>{contentHeader}</div>}
```
Comment (jperals) [inline src/app-layout/skeleton/layout.tsx:67]: We use kebab-case everywhere else, so why camelCase here?

File: src/app-layout/skeleton/layout.tsx [hunk 3]
```diff
@@ -0,0 +1,78 @@
...
+          <div className={styles.contentMain}>{content}</div>
```
Comment (jperals) [inline src/app-layout/skeleton/layout.tsx:68]: This selector doesn't exist in the stylesheet

File: src/app-layout/skeleton/layout.tsx [hunk 4]
```diff
@@ -0,0 +1,78 @@
...
+    | 'tools'
+    | 'toolsWidth'
+    | 'placement'
+  > {
+  topBar?: React.ReactNode;
+}
+
+export function SkeletonLayout({
+  notifications,
+  contentHeader,
```
Comment (jperals) [inline src/app-layout/skeleton/layout.tsx:32]: Just to make sure, we want to support `contentHeader` here although it is deprecated in App Layout?

File: src/app-layout/skeleton/styles.scss [hunk 5]
```diff
@@ -0,0 +1,82 @@
...
+}
+
+.right-panel {
+  grid-area: rightPanel;
+  border-inline-start: 1px solid awsui.$color-border-divider-panel-side;
+}
+
+.top-bar {
+  grid-area: topBar;
+  block-size: 48px;
```
Comment (jperals) [inline src/app-layout/skeleton/styles.scss:59]: What is this number based on?

File: src/app-layout/skeleton/layout.tsx (hunk 2 above)
Comment (just-boris) [reply src/app-layout/skeleton/layout.tsx]: it is easier to type without `['']` and maybe saves some bytes.

But I can also be consistent and rename

File: src/app-layout/skeleton/layout.tsx (hunk 3 above)
Comment (just-boris) [reply src/app-layout/skeleton/layout.tsx]: True, I removed all related styles after several implementation tries.

Cleaned it up here as well. But we keep the `<div>` to ensure any `:first-child` selectors inside the `content` slot do not break

File: src/app-layout/skeleton/styles.scss (hunk 5 above)
Comment (just-boris) [reply src/app-layout/skeleton/styles.scss:59]: Design decision.

File: src/app-layout/skeleton/layout.tsx (hunk 4 above)
Comment (just-boris) [reply src/app-layout/skeleton/layout.tsx:32]: We release this feature as a non-breaking change, so we are required to support existing slots

**Review by jperals — APPROVED** [review]

**Pull request merged**